src/snowflake/connector/connection.py

```diff
@@ -1981,3 +1981,23 @@ def _log_telemetry_imported_packages(self) -> None:
                     connection=self,
                 )
             )
+
+    def is_valid(self) -> bool:
+        """This function tries to answer the question: Is this connection still good for sending queries?
+        Attempts to validate the connections both on the TCP/IP and Session levels."""
+        logger.debug("validating connection and session")
+        if self.is_closed():
+            logger.debug("connection is already closed and not valid")
+            return False
+        else:
```

Comment:
You could get rid of this line and dedent the rest of the function

Reply:
good one - and done